cmd/edit/addon/cmd.go

```diff
@@ -125,16 +131,35 @@ func run(cmd *cobra.Command, argv []string) {
 		if addOnInstallationParam != nil && !param.Editable() {
 			return true
 		}
-
 		var val string
 		var options []string
 		var values []string
 
 		parameterOptions, _ := param.GetOptions()
 
 		for _, opt := range parameterOptions {
-			options = append(options, opt.Name())
-			values = append(values, opt.Value())
+			if param.ValueType() != "resource" {
+				options = append(options, opt.Name())
+				values = append(values, opt.Value())
+				continue
+			}
+
+			paramValueInt, err := strconv.Atoi(addOnInstallationParam.Value())
+			if err != nil {
+				r.Reporter.Errorf("%s", err)
+				os.Exit(1)
+			}
+			// available quota == what's allowed minus what's consumed plus the value of the installation being edited
+			availableQuota := quotaCost.Allowed() - quotaCost.Consumed() + paramValueInt
+			valueInt, err := strconv.Atoi(opt.Value())
+			if err != nil {
+				r.Reporter.Errorf("%s", err)
+				os.Exit(1)
+			}
+			if valueInt <= availableQuota {
+				options = append(options, opt.Name())
+				values = append(values, opt.Value())
+			}
```

Comment:
The more I am thinking about this, we are assuming the quota does not change between when we fetch the quota and the params, versus when the user actually sends the request. 

I think we should maintain the current logic, give the user all options and allow them to make the request, and let AMS make the decision to reject or accept it. 

WDYT